pkg/install/install.go

```diff
@@ -315,3 +347,64 @@ func (i *Installer) initializeKubernetesClients(ctx context.Context) error {
 	i.configcli, err = configclient.NewForConfig(restConfig)
 	return err
 }
+
+func (i *Installer) bootstrapConfigMapReady() (bool, error) {
```

Comment:
please pull the functions below into a separate file, perhaps called ready.go or condition.go.

Reply:
please open (and work, if you like) the following follow-up issue: all these functions are kind-of weird and nasty because they swallow their errors.  We can now make them behave normally and centralise the error tolerance in a nested func in the single `wait.PollImmediateUntil` above.  Please do so.

Reply:
Moved condition functions to a separate file and opened https://github.com/Azure/ARO-RP/issues/229.